source/server/listener_manager_impl.h

```diff
@@ -100,9 +105,15 @@ class ProdListenerComponentFactory : public ListenerComponentFactory,
   createDrainManager(envoy::config::listener::v3::Listener::DrainType drain_type) override;
   uint64_t nextListenerTag() override { return next_listener_tag_++; }
 
+  Filter::TcpListenerFilterConfigProviderManagerImpl& getTcpListenerConfigProviderManager() {
```

Comment:
Could this be added to `ListenerComponentFactory` so we can avoid dynamic down-casts?

Reply:
ACK